src/sentry/utils/kvstore/bigtable.py

```diff
@@ -0,0 +1,287 @@
+import enum
+import logging
+import struct
+import zlib
+from datetime import timedelta
+from threading import Lock
+from typing import Any, Callable, Iterator, Mapping, NamedTuple, Optional, Sequence, Tuple, cast
+
+import zstandard
+from django.utils import timezone
+from google.api_core import exceptions, retry
+from google.cloud import bigtable
+from google.cloud.bigtable.row_data import PartialRowData
+from google.cloud.bigtable.row_set import RowSet
+from google.cloud.bigtable.table import Table
+from sentry.utils.kvstore.abstract import KVStorage
+
+
+logger = logging.getLogger(__name__)
+
+
+class CompressionStrategy(NamedTuple):
+    compress: Callable[[bytes], bytes]
+    decompress: Callable[[bytes], bytes]
```

Comment:
This could also be an actual class (or codec), but I thought the names were significant enough to give it some structure beyond `Tuple[Callable[[bytes], bytes], Callable[[bytes], bytes]]`. (More on this later.)